src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -129,7 +140,12 @@ internal Binder Binder
             SyntaxToken identifierToken,
             CSharpSyntaxNode context)
         {
-            return new OutLocalSymbol(containingSymbol, scopeBinder, enclosingBinderOpt, typeSyntax, identifierToken, context);
+            if (typeSyntax.IsVar)
+            {
```

Comment:
Unlike this code, MakeLocal does not check the form of the type syntax. It distinguishes which subclass to construct based on whether the syntax of the construct contains a context (an initializer) from which a type can be inferred. Out locals always have a context from which a type can be inferred. So the design is inconsistent.